runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -3142,6 +3147,11 @@ done:;
 				}
 				objectClass = VM_VMHelpers::currentClass(objectClass);
 			}
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
```

Comment:
This is likely unnecessary, as the object will be completely overwritten by the copy.